_posts/2023-08-14-clear-specification-and-implementation.markdown

```diff
@@ -0,0 +1,610 @@
+---
+layout: post
+title:  "Clear Separation of Specification and Implementation in Dafny"
+date:   2023-08-14 00:00:00 +0100
+author: Aaron Tomb
+categories:
+---
+
+Dafny allows you to describe computations in several ways,
+primarily pure functions and imperative methods, each with its
+own tradeoffs. Because of these tradeoffs, it can be useful to
+combine mechanisms and relate them to each other. This post
+describes one particularly effective technique for doing so, by
+clearly separating a _specification_ from an _implementation_,
+proving properties of the specification, and proving equivalence (or
+refinement) between the specification and the implementation.
+
+To get started, let's go over the tools Dafny provides. The two most
```

Comment:
To get started, it would help if you could state an *example* quickly enough of what the problem is about.